art/classifiers/GPy.py

```diff
@@ -0,0 +1,254 @@
+        :type model: `sklearn`
+        :param channel_index: Index of the axis in data containing the color channels or features. Not used in this
+               class.
+        :type channel_index: `int`
+        :param defences: Defences to be activated with the classifier.
+        :type defences: :class:`.Preprocessor` or `list(Preprocessor)` instances
+        :param preprocessing: Tuple of the form `(subtractor, divider)` of floats or `np.ndarray` of values to be
+               used for data preprocessing. The first value will be subtracted from the input. The input will then
+               be divided by the second one.
+        :type preprocessing: `tuple`
+        """
+        super(GPyClassifier, self).__init__(clip_values=clip_values,
+                                            defences=defences, preprocessing=preprocessing)
+
+        self.model = model
+        self._input_shape = None
+
+    def fit(self, x, y, batch_size=128, nb_epochs=20, **kwargs):
+        """
+        Fit the classifier on the training set `(x, y)`.
+
+        :param x: Training data. Not used, as goven to model in initialized earlier.
+        :type x: `np.ndarray`
+        :param y: Labels, one-vs-rest encoding. Not used, as goven to model in initialized earlier.
+        :type y: `np.ndarray`
+        :param batch_size: Size of batches. Not used in this class
+        :type batch_size: `int`
+        :param nb_epochs: Number of iterations to be optimized.
+        :type nb_epochs: `int`
+        :type kwargs: `dict`
+        :return: `None`
+        """
+        raise NotImplementedError
+
+    def predict(self, x, logits=False, batch_size=128):
+        """
+        Perform prediction for a batch of inputs.
+
+        :param x: Test set.
+        :type x: `np.ndarray`
+        :param logits: `True` if the prediction should be done at the logits layer.
+        :type logits: `bool`
+        :param batch_size: Size of batches. Not used in this function.
+        :type batch_size: `int`
+        :return: Array of predictions of shape `(nb_inputs, self.nb_classes)`.
+        :rtype: `np.ndarray`
+        """
+        raise NotImplementedError
+
+    def predict_uncertainty(self, x, logits=False, batch_size=128):
+        """
+        Perform uncertainty prediction for a batch of inputs.
+
+        :param x: Test set.
+        :type x: `np.ndarray`
+        :param logits: `True` if the prediction should be done at the logits layer.
+        :type logits: `bool`
+        :param batch_size: Size of batches. Not used in this function.
+        :type batch_size: `int`
+        :return: Array of uncertainty predictions of shape `(nb_inputs, self.nb_classes)`.
+        :rtype: `np.ndarray`
+        """
+        raise NotImplementedError
+
+    def save(self, filename, path=None):
+        self.model.save_model(filename, save_data=False)
+
+
+class GPyGaussianProcessClassifier(GPyClassifier, ClassifierGradients):
```

Comment:
Would it be possible to combine the two classes in this module into 1 class or are they split into 2 classes for a future extension to include `sklearn.gaussian_process.GaussianProcessClassifier`?